builder/parser/line_parsers.go

```diff
@@ -195,6 +194,35 @@ func parseLabel(rest string) (*Node, map[string]bool, error) {
 	return parseNameVal(rest, "LABEL")
 }
 
+// parses a statement with multiple keywords definition and name-value
+// assignment, like `name1 name2 name3=value`.
+// Note that this is a stricter format than the old format of assignment, in a way
+// that it only allows assignment of the form `name=value`. And that a name
+// definition or assignment to an empty value is designated by `name`.
```

Comment:
If the `=` sign is present as in `ARG key=` then we treat it same as `ARG key=""` i.e. it defines a keyword and assigns it an empty value. But `ARG key` just defines a name and doesn't assigns a value as you pointed out in your comment before.

Is the following rewording more clear?

```
// parses a statement containing one or more keyword definition(s) and/or 
// value assignments, like `name1 name2= name3="" name4=value`.
// Note that this is a stricter format than the old format of assignment, in a way
// that it only allows assignment of the form `name4=value` above. In addition, a keyword
// definition alone is of the form `name1` above. And the assignments `name2=` and
// `name3=""` are equivalent and assign an empty value to the respective keywords.
func parseNameOrNameVal(rest string) (*Node, map[string]bool, error)
```